CMakeLists.txt

```
@@ -0,0 +1,265 @@
+#[[
+AUTHOR(S):  Rashad Kanavath <rashad km gmail>
+PURPOSE:    GRASS GIS root that adds options to activate/deactivate
+            3rd party libraries
+COPYRIGHT:  (C) 2020-2022 by the GRASS Development Team
+            This program is free software under the GPL (>=v2)
+            Read the file COPYING that comes with GRASS for details.
+#]]
+
+cmake_minimum_required(VERSION 3.11)
+set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)
+
+# if(DEFINED ENV{VCPKG_DEFAULT_TRIPLET} AND NOT DEFINED VCPKG_TARGET_TRIPLET)
+# set(VCPKG_TARGET_TRIPLET "$ENV{VCPKG_DEFAULT_TRIPLET}" CACHE STRING "")
+# endif()
+
+project(GRASSGIS)
```

```suggestion
project(GRASS)
```
Was there a reason that the project name had to be other than GRASS. The top level call to project defines the name for many many variables, including others that are used like `GRASS_VERSION_MAJOR`.
In the CMake docs, they are prefixed `<PROJECT-NAME>`.

---

@echoix  No need to review this yet, very much WIP, should probably be converted to draft.

---

No reason. ;-)